File: ios/RnDetectDeveloperOptions.m
#import <React/RCTBridgeModule.h>
#import <UIKit/UIKit.h>

@interface RCT_EXTERN_MODULE(RnDetectDeveloperOptions, NSObject)

RCT_EXTERN_METHOD(isDeveloperOptionsEnabled:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/RnDetectDeveloperOptions.swift
import Foundation

@objc(RnDetectDeveloperOptions)
class RnDetectDeveloperOptions: NSObject {

  @objc
  func isDeveloperOptionsEnabled(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    #if targetEnvironment(simulator)
        resolve(true)
    #else
        resolve(false) // No direct detection on physical device
    #endif
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
